### DISCUSSION TITLE
useEffect not console logging - back to preschool

### DISCUSSION THREAD

**Discussion by DrillableBit** [vercel/next.js#45699] [Help] [answered]
Hello there!

I'm trying to console.log data inside a useEffect. 
"Outside useEffect" gets logged to VSCode terminal after client and server is compiled, and page is successfully rendered in browser. 
However, all other console logs are omitted - both in VSC and browser. 

Have I missed some additional configuration? 
Running React v18.2.0 and Next v13.1.2 

````
import React, { useState, useEffect } from "react";

export default function Home({ allPostsData, allProductData }) {

  console.log("Outside useEffect");
 
  const FunctionLog = async () => {
    console.log("Inside useEffect");
  };

  useEffect(() => {
    console.log("Before function")
    FunctionLog();

  }, []);

  return ( .....................
````

**Comment by icyJoseph** [comment]
This should work, although logs within useEffect's stack call will only show up in the browser.

Are you sure you've got things running correct? Maybe you have started the project twice? Have you silenced logs?

It the page actually mounted?

**Reply by DrillableBit** [reply]
Hi,
No console logs in the browser (Chrome). 
I'm currently looking at how to silence logs, but it's not something I've done explicitly.

Page is mounted, features like SWR works on rendered page.

**Reply by DrillableBit** [reply]
Unless mounted means - npm run build 'ing the page?

**Reply by icyJoseph** [reply]
Nope, building has nothing to do here.

I think we are better off if you share a link to a repository. I wonder if you, through not fault of your own, are omitting details that are important to crack this mystery. It is better to see code.

**Reply by DrillableBit** [reply]
Certainly, one sec

**Reply by DrillableBit** [reply]
https://github.com/DrillableBit/using-efkt

**Reply by icyJoseph** [reply]
A little pro-tip: 

You can open those directly up in Stackblitz: https://stackblitz.com/github/DrillableBit/using-efkt?file=src%2Fpages%2Findex.js

What I see, locally, it runs well, but on Stackblitz it sort of hangs... Are you running it on Stackblitz?

**Reply by icyJoseph** [reply]
Also, could you try with incognito mode? I wonder if you can run the app from another terminal. Perhaps VSCode's built in is making things a bit more difficult that they should be.

**Comment by ciruz** [comment]
Hey,

I tested your code with next.js 13 and I get this console output in my browser:

![image](https://user-images.githubusercontent.com/987947/217530012-04a219f8-2a33-413f-9999-d55a395948d8.png)

If the component is a Client Component you get no output in the terminal (vs code), also ``useEffect`` don't run on the Server, it only runs on the Client side.

If you put a ``console.log`` in a Server Component you get output only in the Terminal.

**Reply by DrillableBit** [reply]
Hi, 

The code in question needs to be run on the client side. 

Do you have to run "npm run build" to see client side useEffect in the browser? 

If you run npx create-next-app <name>
... and add a useEffect before the render inside the default component - is that considered client side?

**Reply by ciruz** [reply]
No I just started it with yarn dev / npm run dev.

**Comment by DrillableBit** [comment]
https://github.com/DrillableBit/using-efkt

**Reply by ciruz** [reply]
Looks good to me:

![image](https://user-images.githubusercontent.com/987947/217541593-1f61099e-b2b2-47f6-929a-55d470301d87.png)


This runs on Client and Server, so the output is in the Terminal and also in Browser console:
https://github.com/DrillableBit/using-efkt/blob/main/src/pages/index.js#L7

This is a Client hook, it only runs on the Client Side, so the output is only in the Browser:
https://github.com/DrillableBit/using-efkt/blob/main/src/pages/index.js#L9

**Reply by DrillableBit** [reply]
Thank you! How unfortunate. 

I will try updating Chrome and VScode, it's weird that this behavior only affects Next and not create-react-app....

**Reply by ciruz** [reply]
Also make sure you have the correct node version for next 13.

**Reply by DrillableBit** [reply]
It did affect create-react-app.

**Comment by DrillableBit** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/94070914/217545191-855c3759-939d-4b3f-baec-9305d73aaa9d.png)

Humongous facepalm, but in case it helps someone down the line. 
The browser has a persistent filter.

Thanks to everyone who helped me figure it out.

**Reply by icyJoseph** [reply]
😅 it happens! Sort-of silenced your own logs

**Reply by DrillableBit** [reply]
Yeah haha, the programmer instinct was right from the start. I will make sure to check out stackblitz. Thanks!
